### PR TITLE
Code 194

### PR DESCRIPTION
**Pull request by AkagiShigeru**
Code-191 PR should be merged first and this one should afterward be rebased on top of main.

Changes here were implemented because the menu creation failed if there were either no manual ingredients or recipes at all in the menu list. The code was slightly rewritten for compactness (logic unchanged).

### CODE REVIEW

File: sous-chef/sous_chef/menu/create_menu.py [hunk 1]
```diff
@@ -68,7 +69,9 @@ class Menu:
     dataframe: pd.DataFrame = None
 
     def finalize_fixed_menu(self, gsheets_helper: GsheetsHelper):
-        self.dataframe = self._load_fixed_menu(gsheets_helper)
+        self.dataframe = self._load_fixed_menu(gsheets_helper).reset_index(
```
Comment (ArBridgeman) [inline sous-chef/sous_chef/menu/create_menu.py:72]: Why is this change needed?

File: sous-chef/sous_chef/menu/create_menu.py [hunk 2]
```diff
@@ -78,18 +81,18 @@ def get_menu_for_grocery_list(
     ) -> (list[MenuIngredient], list[MenuRecipe]):
         self.load_local_menu()
 
-        manual_ingredient_list = (
-            self.dataframe[self.dataframe["type"] == "ingredient"]
-            .apply(self._retrieve_manual_menu_ingredient, axis=1)
-            .tolist()
-        )
+        entry_funcs = {
+            "ingredient": self._retrieve_manual_menu_ingredient,
```
Comment (ArBridgeman) [inline sous-chef/sous_chef/menu/create_menu.py:85]: It could stay as it is, but what about moving this to a function. It could be that one exists that could be adapted to handle this.

File: sous-chef/sous_chef/menu/create_menu.py [hunk 3]
```diff
@@ -78,18 +81,18 @@ def get_menu_for_grocery_list(
...
+                    self.dataframe[mask].apply(entry_fct, axis=1).tolist()
+                )
 
-        recipe_list = (
-            self.dataframe[self.dataframe["type"] == "recipe"]
-            .apply(self._retrieve_menu_recipe, axis=1)
-            .tolist()
-        )
-        return manual_ingredient_list, recipe_list
+        return result_dict["ingredient"], result_dict["recipe"]
```
Comment (ArBridgeman) [inline sous-chef/sous_chef/menu/create_menu.py:95]: How does this better handle when these are not set?

**Review by ArBridgeman — CHANGES_REQUESTED** [review]

File: sous-chef/sous_chef/formatter/units.py [hunk 4]
```diff
@@ -34,6 +35,7 @@
     unit_registry.pinch,
     unit_registry.sheet,
     unit_registry.slice,
+    unit_registry.strip,
```
Comment (ArBridgeman) [inline sous-chef/sous_chef/formatter/units.py:38]: Should be able to rebase now

File: sous-chef/sous_chef/menu/create_menu.py (hunk 2 above)
Comment (AkagiShigeru) [reply sous-chef/sous_chef/menu/create_menu.py:85]: Yes, thought about this too, I think it would be better, I'll take a look.
